src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8181,52 +8181,302 @@ public class Test : ITest
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
-        public async Task GenericInterfaceNotNull1()
+        public async Task TestFlowAnalysisAttributeOnNonNullableReferenceType()
         {
-            await TestInRegularAndScriptAsync(
-@$"#nullable enable 
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : [|IInterface<string>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string>
+{{
+    public string MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    public string MethodDisallowNull(string value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNullableReferenceType()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : [|IInterface<string?>|]
+{{
+}}",
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string?>
+{{
+    public string? MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    [return: NotNull]
+    public string? MethodDisallowNull([DisallowNull] string? value)
```

Comment:
❕ Return type is unnecessarily complicated

```suggestion
    public string MethodDisallowNull([DisallowNull] string? value)
```